main.c

```diff
@@ -29,21 +29,27 @@
 #define NETWORK_PREFIX_MIN (1)
 // >=26 is not supported because the recommended guest IP is set to network addr + 100 .
 #define NETWORK_PREFIX_MAX (25)
+#define NS_NAME_MAX (32)
 
-static int nsenter(pid_t target_pid)
+static int nsenter(pid_t target_pid, char *netns, char *userns)
 {
-	char userns[32], netns[32];
 	int usernsfd, netnsfd;
-	snprintf(userns, sizeof(userns), "/proc/%d/ns/user", target_pid);
-	snprintf(netns, sizeof(netns), "/proc/%d/ns/net", target_pid);
-	if ((usernsfd = open(userns, O_RDONLY)) < 0) {
-		perror(userns);
-		return usernsfd;
+	if (!netns) {
+		netns = realloc(netns, NS_NAME_MAX);
```

Comment:
nit: can be just malloc

Reply:
where are these freed?

Reply:
you should also check if it returns NULL.

In general it will be safer to just use `asprintf(&netns...)` so to not allocate more memory than necessary